mongodbatlas/data_source_mongodbatlas_alert_configuration.go

```diff
@@ -504,3 +507,178 @@ func convertNotificationToCtyValues(notification *matlas.Notification) map[strin
 
 	return values
 }
+
+func flattenAlertConfigurationMetricThreshold(m *matlas.MetricThreshold) map[string]interface{} {
+	if m != nil {
+		return map[string]interface{}{
+			"metric_name": m.MetricName,
+			"operator":    m.Operator,
+			"threshold":   cast.ToString(m.Threshold),
+			"units":       m.Units,
+			"mode":        m.Mode,
+		}
+	}
+
+	return map[string]interface{}{}
+}
```

Comment:
why are you still using `map[string]interface` instead of go struct?

Reply:
more detail in comment below